connmgr/connmanager.go

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2016 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package connmgr
+
+import (
+	"fmt"
+	mrand "math/rand"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/addrmgr"
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/wire"
+)
+
+var (
+	//MaxOutboundPeers is the number of max outbound connections.
+	MaxOutboundPeers = 8
+
+	// MaxConnections is the number of max connections.
+	MaxConnections = 125
+
+	// Lookup looks up the host using the local resolver.
+	Lookup = net.LookupIP
+
+	// Dial connects to the address on the named network.
+	Dial = net.Dial
+
+	// ChainParams identifies the chain params to use.
+	ChainParams = &chaincfg.MainNetParams
+)
+
+// ConnResult handles the result of an Connect request.
+type ConnResult struct {
+	Addr      string
+	Permanent bool
+	Conn      net.Conn
+	Err       error
+}
+
+// ConnManager provides a generic connection manager for the bitcoin network.
+type ConnManager struct {
+	wg               sync.WaitGroup
+	newConnections   chan *ConnResult
+	closeConnections chan string
+	wakeup           chan struct{}
+	quit             chan struct{}
+
+	amgr           *addrmgr.AddrManager
+	connections    map[string]net.Conn
+	outboundGroups map[string]int
+
+	NewConnectionHandler func(<-chan *ConnResult)
+}
+
+// seedFromDNS uses DNS seeding to populate the address manager with peers.
+func (cm *ConnManager) seedFromDNS(DNSSeeds []string) {
+	for _, seeder := range DNSSeeds {
```

Comment:
Once the `ConnManager` has a `Config` instance set in the state as mentioned elsewhere, this would just become:

```Go
for _, seeder := range cm.cfg.DNSSeeds {
...
}
```